src/acp/rule.test.ts

```diff
@@ -0,0 +1,704 @@
+/**
+ * Copyright 2020 Inrupt Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to use,
+ * copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the
+ * Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+ * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+ * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+import { jest, describe, it, expect } from "@jest/globals";
+import { asIri, createThing } from "../thing/thing";
+import {
+  addForbiddenRuleToPolicy,
+  addOptionalRuleToPolicy,
+  addRequiredRuleToPolicy,
+  getForbiddenRuleOnPolicyAll,
+  getOptionalRuleOnPolicyAll,
+  getRequiredRuleOnPolicyAll,
+  Rule,
+  setForbiddenRuleOnPolicy,
+  setOptionalRuleOnPolicy,
+  setRequiredRuleOnPolicy,
+} from "./rule";
+import { DataFactory } from "n3";
+import { Thing, ThingPersisted } from "../interfaces";
+import { Policy } from "./policy";
+
+const ACP_ANY = "http://www.w3.org/ns/solid/acp#anyOf";
+const ACP_ALL = "http://www.w3.org/ns/solid/acp#allOf";
+const ACP_NONE = "http://www.w3.org/ns/solid/acp#noneOf";
+
+const mockRule = (url: string) =>
+  createThing({
+    url,
+  });
+
+const addAll = (
```

Comment:
Could this helper function be more clearly named as `addRulesToThing()`  (and the 3rd param typed as a `Rule`) - as that seems to be the intent of the function...?

Reply:
The function name is changed in https://github.com/inrupt/solid-client-js/pull/525. Even if it is indeed only used here to add rules to policy, it is actually more generic than that. Should the name reflect how it is used here, or what it does in general ?

Reply:
I'd always go with how it's intended to be used - otherwise you might wrongly assume it is and will remain more generic than it is, re-use it elsewhere in the codebase, and then suddenly break that if you modify it for the use case here.

Reply:
That makes sense. I'll rename these in the next PR, to avoid conflicts.